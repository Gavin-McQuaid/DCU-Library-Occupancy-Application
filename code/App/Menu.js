import React, {Component} from 'react';
import {Text,View, TouchableHighlight, Alert,AppRegistry ,StyleSheet, Image,KeyboardAvoidingView,TouchableOpacity,FlatList} from 'react-native';
import { ListItem } from 'react-native-elements'
import { NavigationActions } from 'react-navigation';
import {Ionicons} from '@expo/vector-icons';
//Comment line above and uncomment line below if you want to run integration tests using Cavy and the iOS simulator
//import Ionicons from 'react-native-vector-icons/Ionicons';
import { hook} from 'cavy';
const resetAction = NavigationActions.reset({
  index: 0,
  actions: [NavigationActions.navigate({ routeName: 'Login', screen: 'Login' })],
});

class Menu extends Component{
	static navigationOptions = {
	title : 'Menu'
	}
	
	render() {
		return (
		<View style = {styles.container}>
			<View style = {styles.topBar}
			 	ref = {this.props.generateTestHook('Menu.topBar')}> 
				<Ionicons 
					name="md-settings" 
					size={40} 
					color="#FFFFFF"
					onPress={() => this.props.navigation.navigate("Settings", 
					{screen: "Settings", email: this.props.navigation.state.params.email })}
				/>
			</View>
			<View style = {styles.logoContainer}>
					<Text style = {styles.title}> DCU Library App 
					</Text>
					<Image
						style = {styles.logo}
						source = {require('./images/logo.png')} 
					/>			
			</View>
			<FlatList 
				style = {styles.Flat}
				scrollEnabled={false}
  				data={[{ key: 'Basement', image: 'basement.png'},
  					   { key: 'Ground Floor', image: 'ground.png'},
  					   { key: 'First Floor', image: 'first.png'},
   					   { key: 'Second Floor', image: 'second.png'}]}
   				renderItem={({item}) => 
   				<ListItem containerStyle= 
  					{{padding:0,margin:0, flex:1, backgroundColor : 'cornflowerblue', paddingTop: 15}} 
  					title={item.key} 
  					titleStyle = {styles.titles}
  					rightIcon={<Ionicons 
  								name="ios-arrow-dropright" 
  								size={20} 
  								color="#FFFFFF"
  							   />}
  					onPress={() => this.props.navigation.navigate("Floor", {screen: "Floor", image: item.image, 
  					email: this.props.navigation.state.params.email, 
  					floor: item.key  })}
  				/>}
			/>	
		</View>		
		);
	}
}
export default hook(Menu);
const styles = StyleSheet.create({
	container: {
		backgroundColor : 'cornflowerblue',  
		flex:1,
		justifyContent : 'center'
	},

	topBar: {
		backgroundColor: 'cornflowerblue',  
		height: 50,
		alignSelf: 'flex-end',
		paddingRight: 3,
	},
	
	Flat: {
		paddingTop: 10,
		margin:0,
		flex: 1,
		backgroundColor : 'cornflowerblue',	
    },

  	titles: {
  		color: 'white',
  		fontSize: 22,
  		justifyContent: 'center'
  	},

    logoContainer: {
		alignItems : 'center',
		justifyContent : 'center'
    },

    logo: {
		width : 200,
		height : 150
    },

    title: {
		color : 'white',
		marginBottom : 10,
		fontSize : 30,
		opacity : 0.9,
		justifyContent: 'center'
	}
  
});