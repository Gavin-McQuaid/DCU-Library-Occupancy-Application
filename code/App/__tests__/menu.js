import React from 'react';
import Menu from '../Menu';
import renderer from 'react-test-renderer';
import {shallow, mount,render,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockStore from 'redux-mock-store';
const Text = require('Text');
const TouchableHighlight = require('TouchableHighlight');
const ReactTestRenderer = require('react-test-renderer');
const TouchableOpacity = require('TouchableOpacity');
const TextInput = require('TextInput');
const Image = require('Image');
import { ListItem } from 'react-native-elements'

import Icon from 'react-native-vector-icons/FontAwesome';
const FlatList = require('FlatList');
var ReactTestUtils = require('react-dom/test-utils')
configure({ adapter: new Adapter() })
describe('Menu Page', () =>{
	it( 'icon shows', () => {
		const instance0 = ReactTestRenderer.create(
		<Icon name = 'cog'></Icon>
		);
	});	
	it ('list shows',() => {
		const instance1 = ReactTestRenderer.create(
		<FlatList
		data={[{ key: 'Basement', image: 'basement.png'},
{ key: 'Ground Floor', image: 'ground.png'},
  { key: 'First Floor', image: 'first.png'},
  { key: 'Second Floor', image: 'second.png'}]}
  renderItem={({item}) => <ListItem title={item.key} />}
		/>
		);
	});
	it('Flatlist itemworks', () => {
		const wrapper = shallow(<Menu />);
		const Item1 = wrapper.find('FlatList');
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<Menu store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
	const rendered = renderer.create(<Menu/>).toJSON();
		expect(rendered).toMatchSnapshot();
});


