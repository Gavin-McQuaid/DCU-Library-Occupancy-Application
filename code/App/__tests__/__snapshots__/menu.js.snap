// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "cornflowerblue",
        "height": 50,
        "paddingRight": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "justifyContent": "center",
          "marginBottom": 10,
          "opacity": 0.9,
        }
      }
    >
       DCU Library App
    </Text>
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 150,
          "width": 200,
        }
      }
    />
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "image": "basement.png",
          "key": "Basement",
        },
        Object {
          "image": "ground.png",
          "key": "Ground Floor",
        },
        Object {
          "image": "first.png",
          "key": "First Floor",
        },
        Object {
          "image": "second.png",
          "key": "Second Floor",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
        "margin": 0,
        "paddingTop": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Basement
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Ground Floor
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  First Floor
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Second Floor
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Menu Page Flatlist itemworks 1`] = `
<View
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "cornflowerblue",
        "height": 50,
        "paddingRight": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "justifyContent": "center",
          "marginBottom": 10,
          "opacity": 0.9,
        }
      }
    >
       DCU Library App
    </Text>
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 150,
          "width": 200,
        }
      }
    />
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "image": "basement.png",
          "key": "Basement",
        },
        Object {
          "image": "ground.png",
          "key": "Ground Floor",
        },
        Object {
          "image": "first.png",
          "key": "First Floor",
        },
        Object {
          "image": "second.png",
          "key": "Second Floor",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
        "margin": 0,
        "paddingTop": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Basement
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Ground Floor
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  First Floor
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "cornflowerblue",
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "paddingTop": 15,
              },
              false,
              false,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#43484d",
                          "fontSize": 17.5,
                        },
                        Object {
                          "marginLeft": 10,
                        },
                        Object {
                          "color": "white",
                          "fontSize": 22,
                          "justifyContent": "center",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Second Floor
                </Text>
              </View>
              <View
                style={undefined}
              >
                <View />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
